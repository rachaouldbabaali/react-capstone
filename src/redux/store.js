import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import thunkMiddleware from 'redux-thunk';
import weatherReducer from './weatherSlice';

const middleware = [...getDefaultMiddleware(), thunkMiddleware];

const store = configureStore({
  reducer: {
    weather: weatherReducer,
  },
  middleware,
});

export default store;
