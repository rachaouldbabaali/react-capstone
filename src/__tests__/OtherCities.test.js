import React from 'react';
import { useSelector, useDispatch, Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Weather from '../components/Othercities';
import store from '../redux/store';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('Weather component', () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => callback({
      weather: {
        weather: null,
      },
      status: null,
    }));
    useDispatch.mockReturnValue(jest.fn());
  });

  test('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Weather />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
