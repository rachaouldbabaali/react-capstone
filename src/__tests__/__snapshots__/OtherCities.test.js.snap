// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component renders correctly 1`] = `
<div>
  <div
    class="other"
  >
    <div
      class="container"
    >
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            New York
          </div>
          <div
            class="city-map"
          >
            <img
              src="us.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            London
          </div>
          <div
            class="city-map"
          >
            <img
              src="gb.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Paris
          </div>
          <div
            class="city-map"
          >
            <img
              src="fr.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Tokyo
          </div>
          <div
            class="city-map"
          >
            <img
              src="jp.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Dubai
          </div>
          <div
            class="city-map"
          >
            <img
              src="ae.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Sydney
          </div>
          <div
            class="city-map"
          >
            <img
              src="au.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Sao Paulo
          </div>
          <div
            class="city-map"
          >
            <img
              src="br.png"
            />
          </div>
        </div>
      </div>
      <div
        class="city-card card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            New Delhi
          </div>
          <div
            class="city-map"
          >
            <img
              src="in.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
